From stable-bounces@linux.kernel.org Fri Oct 28 07:50:40 2005
Message-ID: <436239DE.6020806@austin.ibm.com>
Date: Fri, 28 Oct 2005 09:46:54 -0500
From: Joel Schopp <jschopp@austin.ibm.com>
To: stable@kernel.org
Cc: 
Subject: [PATCH] ppc64 memory model depends on NUMA

From: Andy Whitcroft <apw@shadowen.org>

Currently when we first select memory model (FLAT, DISCONTIG, SPARSE)
then select whether the machine is NUMA.  However NUMA systems may not
be FLAT.  This constraint it not honoured and we may configure a NUMA/FLAT
system.

Reorder the configuration such that we choose NUMA first which allows us
to only list the memory models which are valid.  We now default NUMA
for known NUMA systems.  Note that this new order also matches that
used in x86.

Signed-off-by: Andy Whitcroft <apw@shadowen.org>
Signed-off-by: Joel Schopp <jschopp@austin.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
diff -upN reference/arch/ppc64/Kconfig current/arch/ppc64/Kconfig
--- reference/arch/ppc64/Kconfig
+++ current/arch/ppc64/Kconfig
@@ -234,6 +234,10 @@ config HMT
  	  This option enables hardware multithreading on RS64 cpus.
  	  pSeries systems p620 and p660 have such a cpu type.

+config NUMA
+	bool "NUMA support"
+	default y if SMP && PPC_PSERIES
+
  config ARCH_SELECT_MEMORY_MODEL
  	def_bool y

@@ -249,9 +253,6 @@ config ARCH_DISCONTIGMEM_DEFAULT
  	def_bool y
  	depends on ARCH_DISCONTIGMEM_ENABLE

-config ARCH_FLATMEM_ENABLE
-	def_bool y
-
  config ARCH_SPARSEMEM_ENABLE
  	def_bool y
  	depends on ARCH_DISCONTIGMEM_ENABLE
@@ -274,10 +275,6 @@ config NODES_SPAN_OTHER_NODES
  	def_bool y
  	depends on NEED_MULTIPLE_NODES

-config NUMA
-	bool "NUMA support"
-	default y if DISCONTIGMEM || SPARSEMEM
-
  config SCHED_SMT
  	bool "SMT (Hyperthreading) scheduler support"
  	depends on SMP
_______________________________________________
Linuxppc64-dev mailing list
Linuxppc64-dev@ozlabs.org
https://ozlabs.org/mailman/listinfo/linuxppc64-dev

_______________________________________________
stable mailing list
stable@linux.kernel.org
http://linux.kernel.org/mailman/listinfo/stable

